Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -599,28 +597,26 @@ namespace Multiplayer
                 // Route to spawner implementation
                 MultiplayerAgentDatum datum;
                 datum.m_agentType = MultiplayerAgentType::Client;
-                controlledEntity = spawner->OnPlayerJoin(packet.GetTemporaryUserId(), datum);
+                auto preActivationCallback = [connection](const INetworkEntityManager::EntityList& entityList)
+                {
+                    EnableAutonomousControl(entityList, connection->GetConnectionId());
+                };
+
+                controlledEntity = spawner->OnPlayerJoin(packet.GetTemporaryUserId(), datum, preActivationCallback);
+                if (controlledEntity.Exists())
+                {
+                    StartServerToClientReplication(packet.GetTemporaryUserId(), controlledEntity, connection);
+                }
+                else
+                {
+                    // If there wasn't any spawner available, wait until a level loads and check again
+                    m_playersWaitingToBeSpawned.push_back(PlayerWaitingToBeSpawned{ packet.GetTemporaryUserId(), datum, connection });
```

Comment:
Tip: you can do use emplace_back and temporary skip object creation in such cases.

Reply:
thank you!